### DISCUSSION TITLE
Jekyll plugin

### DISCUSSION THREAD

**Discussion by fauno** [jameslittle230/stork#96] [Show and tell]
Hi! Thanks for your work! We've been looking for something like this :) We're currently using lunrjs but indexing uses a lot of RAM and time, and we've had some issues with Spanish stemming. Stork just indexed 1100 pages in 2s. We're looking forward to further test stork!

I'm opening this issue because I think it would be interesting to build a Jekyll plugin that can:

* Dynamically generate a config.toml with all paths, URLs and titles
* Run `stork --build` over the generated site
* Store the index on the site

I think I could write a first version in a couple hours :)

Edit: Sorry I forget ctrl+enter sends the issue

**Comment by itamarst** [comment]
Here is a `config.toml` file that renders via `jekyll build`; just drop it in your Jekyll site, do `jekyll build`, and then you can do `stork --build _site/config.toml`. It only does posts, but you can easily add all pages by adding a for loop `for page in site.pages`.

```
---
layout: none
---
[input]
files = [
{% for post in site.posts %}{path = "{{ post.path }}", url = "{{ post.url }}", title = "{{ post.title | replace: '"', '\\"'}}"},
{% endfor %}
]

[output]
filename = "search.st"
```

**Comment by chmaynard** [comment]
This looks great. When this plugin executes, has the Liquid code in each post already been rendered into HTML?

**Comment by itamarst** [comment]
No, you'll notice the path it's including is the path to the original (presumably markdown) post. You could change it to point at the rendered HTML instead, however, by changing `{{ post.path }}` to... something else, not quite sure how but should be doable.

**Comment by fauno** [comment]
Well, here it is, would you try it and tell me how it goes? :)

https://rubygems.org/gems/jekyll-stork

@itamarst thanks for the suggestion, I didn't use Liquid but generated a temporary file from Ruby.

I prefered to index the generated pages instead of the source files (`post.path` leads to the source file), but I found #105.  It requires your theme to put content inside a `<main>` tag for every page.

**Comment by chmaynard** [comment]
`$ gem install jekyll-stork`

ERROR:  Error installing jekyll-stork:
	jekyll-stork-0.1.0 requires Ruby version ~> 2.6. The current ruby version is 3.0.0.0.

**Comment by fauno** [comment]
thanks! please upgrade the gem to 0.1.1 :)

**Comment by chmaynard** [comment]
Perfect, thanks.

**Comment by jameslittle230** [comment]
Hi all! This looks like really good work - glad you've found Stork helpful, and I'm really excited to see people building on top of it! I'm just popping in to say that I'm converting this to a Github Discussion instead of an issue -- hope that works for folks.

**Reply by fauno** [reply]
I edited the title because it's not a proposal anymore :)
